exasol/toolbox/util/release/changelog.py

```diff
@@ -50,6 +56,36 @@ def _extract_unreleased_notes(self) -> str:
         unreleased_content += "\n"
         return unreleased_content
 
+    def _prepare_dependency_update(self) -> str:
+        old_dependencies_in_groups = get_dependencies_from_latest_tag()
+        current_dependencies_in_groups = get_dependencies(
+            working_directory=self.source_path
+        )
+
+        content = ""
+        # preserve order of keys from old group
+        groups = list(
+            dict.fromkeys(
+                chain(
+                    old_dependencies_in_groups.keys(),
+                    current_dependencies_in_groups.keys(),
+                )
+            )
+        )
+        for group in groups:
+            old_dependencies = old_dependencies_in_groups.get(group, {})
+            current_dependencies = current_dependencies_in_groups.get(group, {})
+            changes = DependencyChanges(
+                old_dependencies=old_dependencies,
+                current_dependencies=current_dependencies,
+            ).changes
+            if changes:
+                # nicer group heading
```

Comment:
comment unneeded, as it was a note for feedback. for comparison project keeper has nicer headers. for us to do that, we'd need to define the group to prettier name mapping here or better in `poetry_dependencies.py` for use overall. it could be a bit complicated as need to decide how to pass through, like dict[str, (group_details, packages_dict)] or ...?

https://github.com/exasol/project-keeper/blob/main/doc/changes/changes_5.2.2.md
```suggestion
```

Reply:
Currently, the implementation seems to report groups `main` and `dev` which is perfectly acceptable IMHO.